runtime/lua/vim/lsp/handlers.lua

```diff
@@ -28,6 +28,22 @@ local function err_message(...)
   api.nvim_command('redraw')
 end
 
+--- @param params lsp.ShowMessageParams|lsp.ShowMessageRequestParams
+--- @param ctx lsp.HandlerContext
+local function show_message_notification(params, ctx)
```

Comment:
handlers should be prefixed with `on_`. or if we think of this function as "procedural" we can just call it `show_message()` ?



```suggestion
local function on_show_message(params, ctx)
```

Reply:
Yeah I wouldn't consider this a proper handler, it's just the shared utility between the "showMessage" notification/request to invoke `vim.notify`. I'll stick with the current name though as `vim.ui.select` is another way to show the message, the `_notification` bit is just to clarify that it invokes `vim.notify`.